src/coreclr/vm/exinfo.h

```diff
@@ -207,8 +207,38 @@ struct LastReportedFuncletInfo
     uint32_t Flags;
 };
 
-struct ExInfo : public ExceptionTrackerBase
+struct ExInfo
 {
+    struct DAC_EXCEPTION_POINTERS
+    {
+        PTR_EXCEPTION_RECORD    ExceptionRecord;
+        PTR_CONTEXT             ContextRecord;
+    };
+
+    class StackRange
+    {
+    public:
+        StackRange();
+        void Reset();
+        bool IsEmpty();
+        bool IsSupersededBy(StackFrame sf);
+        void CombineWith(StackFrame sfCurrent, StackRange* pPreviousRange);
+        bool Contains(StackFrame sf);
+        void ExtendUpperBound(StackFrame sf);
+        void ExtendLowerBound(StackFrame sf);
+        void TrimLowerBound(StackFrame sf);
+        StackFrame GetLowerBound();
+        StackFrame GetUpperBound();
+        INDEBUG(bool IsDisjointWithAndLowerThan(StackRange* pOtherRange));
+    private:
+        INDEBUG(bool IsConsistent());
+
+    private:
+        // <TODO> can we use a smaller encoding? </TODO>
```

Comment:
Since you are at it, can you please remove this obsolete comment? I don't have any idea why we would want to make it smaller and this comment has been there even before .NET core existed.

Reply:
Actually, since the CI tests are green, let's not touch the change and let's just merge it in.

Reply:
If we drop the old x86 code at some point we may just merge `ExInfo` into `exceptionhandling.h` and delete it then...